R-package/R/utils.R

```diff
@@ -349,21 +349,41 @@ xgb.createFolds <- function(y, k) {
 #' @name xgboost-deprecated
 NULL
 
-#' Do not use \code{\link[base]{saveRDS}} or \code{\link[base]{save}} for long-term archival of
-#' models. Instead, use \code{\link{xgb.save}} or \code{\link{xgb.save.raw}}.
+#' @title Model Serialization and Compatibility
+#' @description
 #'
-#' It is a common practice to use the built-in \code{\link[base]{saveRDS}} function (or
-#' \code{\link[base]{save}}) to persist R objects to the disk. While it is possible to persist
-#' \code{xgb.Booster} objects using \code{\link[base]{saveRDS}}, it is not advisable to do so if
-#' the model is to be accessed in the future. If you train a model with the current version of
-#' XGBoost and persist it with \code{\link[base]{saveRDS}}, the model is not guaranteed to be
-#' accessible in later releases of XGBoost. To ensure that your model can be accessed in future
-#' releases of XGBoost, use \code{\link{xgb.save}} or \code{\link{xgb.save.raw}} instead.
+#' When it comes to serializing XGBoost models, it's possible to use R serializers such as
+#' \link{save} or \link{saveRDS} to serialize an XGBoost R model, but XGBoost also provides
+#' its own serializers with perhaps better compability guarantees and which allow loading
+#' said models in other language bindings of XGBoost.
+#'
+#' Note however that an `xgb.Booster` object might also keep:\itemize{
+#' \item Additional model configuration attributes (accessible through \link{xgb.config}),
+#' which might be used during model fitting but are not used in e.g. `predict`, feature importance,
+#' or plotting methods.
+#' \item Additional R-specific attributes  - e.g. results of callbacks, such as evaluation logs,
+#' which are kept as a `data.table` object, accessible through `attributes(model)$evaluation_log`
+#' if present.
+#' }
+#'
+#' The first ones (configuration attributes) do not have the same compatibility guarantees as
+#' attributes that are set and accessed through \link{xgb.attributes} - that is, such attributes
+#' might be lost after loading the booster in a different XGBoost version, regardless of the
+#' serializer that was used. Note that these are saved when using \link{xgb.save}, but not when
+#' using \link{xgb.save.raw}.
```

Comment:
I don't think either `xgb.save` or `xgb.save.raw` preserves configuration attributes.

@trivialfis Any thoughts?

Reply:
`xgb.save` is the same as `xgb.save.raw`, the former writes to a file while the latter writes to a memory buffer. You are right that they don't save configuration. Only the removed `xgb.serialize` does.

Reply:
See https://github.com/dmlc/xgboost/pull/9924#discussion_r1445104757

Reply:
My bad, I've updated the note.